packages/testing-helpers/src/helpers.js

```diff
@@ -130,3 +130,51 @@ export function oneEvent(element, eventName) {
     element.addEventListener(eventName, listener);
   });
 }
+
+/**
+ * Waits until the given predicate returns a truthy value. Calls and awaits the predicate
+ * function at the given interval time. Can be used to poll until a certain condition is true.
+ *
+ * @example
+ * ```js
+ * import { fixture, waitUntil } from '@open-wc/testing-helpers';
+ *
+ * const element = await fixture(html`<my-element></my-element>`);
+ *
+ * await waitUntil(() => element.someAsyncProperty);
+ * ```
+ *
+ * @param {() => boolean | Promise<boolean>} asyncPredicate - predicate function which is called each poll interval.
+ *   The predicate is awaited, so it can return a promise or the actual value.
+ * @param {string} [message]
+ * @param {{ interval?: number, timeout?: number }} [options]
+ */
+export function waitUntil(asyncPredicate, message, options = {}) {
```

Comment:
I remember some bikeshedding going on around the linguistics of other async method names...

`await waitUntil`

is not quite:

`await aTimeout`

Should this _maybe_ be

`await aPredicate`

Or similr?

Reply:
Mostly I try to pick names that read well when reading the source code. I don't think `aTimeout` is really a good name, but consistency would be nice. 

I think `aPredicate` doesn't really convey the idea that it's polling and waiting. I did some googling, I did see that `waitUntil` is used quite a bit. `until` would make sense too, but I don't want to overload the lit directive.

Open to other suggestions!

Reply:
That's an interesting problem, because `waitUtil` read better before using `async/await`.

```js
waitUntil(..)
  .then(...);
```

Also I think the name expresses very well what the function does. However with the use of `await` looks redundant reading the code `await waitUntil(...)` and names like `await until`, `await till` or event `await to` looks better reading the code, but I think the function name expresses worse what the function does.

IMHO I prefer to use names that expresses well what the function does, although we could have undesirable side effects like `await waitUntil`.

Reply:
I'm not too bothered by `await waituntil` personally

Reply:
^ Same here

Reply:
the param called `asyncPredicate` isn't really async, is it? like, the predicate is accessed synchronously, we're just polling it to add the async behaviour to the test.

So the sig should be `waitUntil(predicate, message?, options?)`

Reply:
The predicate can be both sync or async, we await the predicate as well. I'll call it `syncOrAsyncPredicate` :p